### ISSUE TITLE
action vs. browserAction

### ISSUE THREAD

**Issue by dword-design** [wxt-dev/wxt#490]
### Feature Request

First of all, thanks for this great Nuxt-inspired tool :).

Looks like the lib currently does not abstract away `action` in v3 and `browserAction` in v2. So one would need to define both in `wxt.config.js` and then distinguish between the cases in the scripts. Would be great to just have `browser.action` from `wxt/browser` and `action` in the config. Or am I missing something?

**Comment by aklinker1** [comment]
Are you talking about the case where you're using an action without a popup entrypoint?

If not, see https://wxt.dev/entrypoints/popup.html.

If so, how do you propose letting the extension opt-in to using a page_action instead of a `browser_action`?

**Comment by aklinker1** [comment]
Are you talking about the case where you're using an action without a popup entrypoint?

If not, see https://wxt.dev/entrypoints/popup.html.

If so, how do you propose letting the extension opt-in to using a page_action instead of a `browser_action`?

**Comment by aklinker1** [comment]
Are you talking about the case where you're using an action without a popup entrypoint?

If not, see https://wxt.dev/entrypoints/popup.html.

If so, how do you propose letting the extension opt-in to using a page_action instead of a `browser_action`?

**Comment by aklinker1** [comment]
Are you talking about the case where you're using an action without a popup entrypoint?

If not, see https://wxt.dev/entrypoints/popup.html.

If so, how do you propose letting the extension opt-in to using a page_action instead of a `browser_action`?

**Comment by aklinker1** [comment]
Correct, I for one perfer page actions over browser actions, and think MV3 made a bad decision choosing browser actions over page actions for the new action field. it prevents a cluttered navigation bar with all your extensions trying to take up space when they're not even being used... All that to say I like page actions and would recommend people use them. But I also understand that browser actions are the most equivalent to the new action, so it makes sense to default to the browser action...

But anyways, my understanding of your suggestion is that when I have a config file that looks like this:

```ts
export default defineConfig({
  manifest: {
    action: {},
  },
});
```

it should automatically output a `browser_action` instead while targetting MV2, is that correct? So I'm curious what you'd recommend for opting into converting it to a `popup_action` instead. I have an idea, but if you have one I'd like to hear that first!

**Comment by aklinker1** [comment]
Oh yeah, sorry, `page_action` is what I meant. 

What about letting you provide a manifest like this:

```js
manifest: {
  action: {},
  browser_action: {},
}
```

And WXT automatically strips out any fields that aren't available for the target manifest version?

Which is annoying to write.

Then if someone wants to use the page action instead, they can just use:

```js
manifest: {
  action: {},
  page_action: {},
}
```

And if you exclude both `page_action` and `browser_action`, MV2 defaults to a `browser_action` for compatibility

```js
manifest: {
  action: {},
}
```

This applies to more than just the `action` fields. That way you don't have to use a function to remove fields based on the target manifest version.

```js
manifest: ({ manifestVersion }) => manfiestVersion === 3 ? { action: {} } : { browser_action: {} },
```

**Comment by aklinker1** [comment]
Released in [`v0.17.6`](https://github.com/wxt-dev/wxt/releases/tag/v0.17.6)
